# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 9c3388d86aa3
# Full Hash: 9c3388d86aa3393eca80434ec86e2a824ec38a5f
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-07-19 21:35:33
# Description:
#   Bug 1779036 - Account for the fact that the ended promise might already have been resolved in GetPosition, during shutdown. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D151592
# ==============================================================================

diff -r 43863430aaa7 -r 9c3388d86aa3 dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Tue Jul 19 15:10:00 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Tue Jul 19 15:10:26 2022 +0000
@@ -107,7 +107,7 @@
       // ended promise.
       if (CheckIfEnded()) {
         MOZ_ASSERT(!mAudioSink);
-        mEndedPromiseHolder.Resolve(true, __func__);
+        mEndedPromiseHolder.ResolveIfExists(true, __func__);
       }
     }
     mLastClockSource = ClockSource::SystemClock;
